src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.cpp

```diff
@@ -384,9 +398,6 @@ ReturnCode_t TypeObjectRegistry::get_type_dependencies(
                 case TI_PLAIN_MAP_LARGE:
                     get_indirect_hash_map_dependencies(type_id.map_ldefn(), type_dependencies);
                     break;
-                default:
-                    return eprosima::fastdds::dds::RETCODE_BAD_PARAMETER;
```

Comment:
The default case might be interesting to be kept. The unreachable line is to add a break to a case that has already returned in the previous line.

Reply:
restored in [d3245e1](https://github.com/eProsima/Fast-DDS/pull/4496/commits/d3245e16dd687c62ce397be8018c718828d2bc6e)

Reply:
Not really. Current implementation returns RETCODE_OK instead of RETCODE_BAD_PARAMETER if it falls through the default case (which should not happen) (!)